### PR TITLE
fix : enable to display published news images selected from existing upload for non members - EXO-65111

### PR DESCRIPTION
**Pull request by sofyenne**
This change is going to display the published news images selected from existing upload for non space members.

### CODE REVIEW

File: data-upgrade-news/src/main/java/org/exoplatform/news/upgrade/jcr/PublishedNewsImagesPermissionsUpgradePlugin.java [hunk 1]
```diff
@@ -159,4 +169,21 @@ private String getStringProperty(Node node, String propertyName) throws Reposito
...
   }
+  private void updateNewsImagesPermissions(Node newsNode, Session session) throws RepositoryException {
+    Matcher matcher = Pattern.compile(IMAGE_SRC_REGEX).matcher(getStringProperty(newsNode, "exo:body"));
+    if (matcher.find()) {
+      updateNewsImagesPermissions(newsNode, session, IMAGE_SRC_REGEX);
+    } else {
+      String existingUploadImagesSrcRegex = "src=\"" + CommonsUtils.getCurrentDomain() + "/" + PortalContainer.getCurrentPortalContainerName() + "/" + CommonsUtils.getRestContextName() + "/jcr/?(.+)?\"";
+      updateNewsImagesPermissions(newsNode, session, existingUploadImagesSrcRegex);
+    }
+  }
```
Comment (ahamdi) [inline data-upgrade-news/src/main/java/org/exoplatform/news/upgrade/jcr/PublishedNewsImagesPermissionsUpgradePlugin.java]: This function is unneeded, could we just do sequentially both regex checks ?

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: data-upgrade-news/src/main/java/org/exoplatform/news/upgrade/jcr/PublishedNewsImagesPermissionsUpgradePlugin.java (hunk 1 above)
Comment (sofyenne) [reply data-upgrade-news/src/main/java/org/exoplatform/news/upgrade/jcr/PublishedNewsImagesPermissionsUpgradePlugin.java]: Done !

File: data-upgrade-news/src/main/java/org/exoplatform/news/upgrade/jcr/PublishedNewsImagesPermissionsUpgradePlugin.java [hunk 2]
```diff
@@ -122,28 +126,54 @@ public void processUpgrade(String s, String s1) {
...
-                                                                                     .equals(PLATFORM_USERS_GROUP_IDENTITY))
-                                     .toList()
-                                     .size() > 0;
-        if (!isPublicImage) {
-          // make news images public
-          image.setPermission(PLATFORM_USERS_GROUP_IDENTITY, READ_PERMISSIONS);
-          image.save();
-          imagesCount += 1;
+      } while (matcher.find());
+    } else {
```
Comment (ahamdi) [inline data-upgrade-news/src/main/java/org/exoplatform/news/upgrade/jcr/PublishedNewsImagesPermissionsUpgradePlugin.java]: What if we have both kind of image URLs inside the same body ? shouldn't we check both regex ?

Comment (sofyenne) [reply data-upgrade-news/src/main/java/org/exoplatform/news/upgrade/jcr/PublishedNewsImagesPermissionsUpgradePlugin.java]: done!

**Review by ahamdi — APPROVED** [review]

**Pull request merged**